### ISSUE TITLE
test_hpc_hgs_implicit_hcc failing with astropy dev

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#4086]
Here's a fun new bug. I think this is an upstream astropy issue related to the fact we have an observer frame attribute that is itself a coordinate. See https://github.com/astropy/astropy/issues/10287 for the upstream issue.

```python
test_hpc_hgs_implicit_hcc ___________________________________________________________________________________

    def test_hpc_hgs_implicit_hcc():
        # An HPC->HGS transformation should give the same answer whether the transformation step
        #   through HCC is implicit or explicit
        start = SkyCoord(0*u.arcsec, 0*u.arcsec, 0.5*u.AU,
                         frame=Helioprojective(obstime='2019-06-01', observer='earth'))
        frame = HeliographicStonyhurst(obstime='2019-12-01')
    
        implicit = start.transform_to(frame)
        explicit1 = start.transform_to(Heliocentric(obstime=start.obstime, observer='earth')).\
            transform_to(frame)
        explicit2 = start.transform_to(Heliocentric(obstime=frame.obstime, observer='earth')).\
            transform_to(frame)
    
        assert_quantity_allclose(implicit.separation_3d(explicit1), 0*u.AU, atol=1e-10*u.AU)
>       assert_quantity_allclose(implicit.separation_3d(explicit2), 0*u.AU, atol=1e-10*u.AU)

sunpy/coordinates/tests/test_transformations.py:500: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../astropy/astropy/coordinates/sky_coordinate.py:1053: in separation_3d
    if not self.is_equivalent_frame(other):
../astropy/astropy/coordinates/sky_coordinate.py:969: in is_equivalent_frame
    if np.any(getattr(self, fattrnm) != getattr(other, fattrnm)):
../astropy/astropy/coordinates/baseframe.py:1656: in __ne__
    return np.logical_not(self == value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <HeliographicStonyhurst Coordinate (obstime=2019-06-01T00:00:00.000): (lon, lat, radius) in (deg, deg, AU)
    (0., -0.72605048, 1.01396311)>
value = <HeliographicStonyhurst Coordinate (obstime=2019-12-01T00:00:00.000): (lon, lat, radius) in (deg, deg, AU)
    (0., 0.94857499, 0.98618233)>

    def __eq__(self, value):
        """Equality operator for frame.
    
        This implements strict equality and requires that the frames are
        equivalent and that the representation data are exactly equal.
        """
        is_equiv = self.is_equivalent_frame(value)
    
        if self._data is None and value._data is None:
            # For Frame with no data, == compare is same as is_equivalent_frame()
            return is_equiv
    
        if not is_equiv:
>           raise TypeError(f'cannot compare: objects must have equivalent frames: '
                            f'{self.replicate_without_data()} vs. '
                            f'{value.replicate_without_data()}')
E           TypeError: cannot compare: objects must have equivalent frames: <HeliographicStonyhurst Frame (obstime=2019-06-01T00:00:00.000)> vs. <HeliographicStonyhurst Frame (obstime=2019-12-01T00:00:00.000)>

../astropy/astropy/coordinates/baseframe.py:1644: TypeError
==================================================================================== short test summary info ====================================================================================
FAILED sunpy/coordinates/tests/test_transformations.py::test_hpc_hgs_implicit_hcc - TypeError: cannot compare: objects must have equivalent frames: <HeliographicStonyhurst Frame (obstime=201...
```

**Comment by ayshih** [comment]
Ugh.  This will require an upstream fix.

**Comment by ayshih** [comment]
Upstream PR has been merged (astropy/astropy#10290)
